### PR TITLE
Handled the case when Maintainer is not a student

### PR DESCRIPTION
**Pull request by GopalGupta8216**
I had done it according to my understanding of the issue. That we were unable to verify any information if the user is not a student , so i had given all privileges to a maintainer who is not a student

### CODE REVIEW

**Comment by GopalGupta8216** [comment]
@pradumangoyal  please checkout

**Comment by pradumangoyal** [comment]
Read the description of issue #10. Rather than giving the privilege blindly, we can have a callback named bool parameter passed in the function itself which is returned if not Student and defaults to `False`. So this will not raise a surprise to current implementations.

**Comment by GopalGupta8216** [comment]
but in this case we are not giving the permissions blindl we are firstl validating whether the user is a maintainer or not, and if he/she is a maintainer than there are only two cases whether he is a student or not. And we are handling both of those cases.

**Comment by GopalGupta8216** [comment]
Furthermore I thought if we although make a function for validating there would be code repetition of get_roles function + we need to change the same thing all other places where we are using get_rights_for_user. @pradumangoyal

**Comment by pradumangoyal** [comment]
Default value handles the case, you need not pass a named parameter compulsory. If you set it to False by default, if the parameter is not passed it will accept False. 

About giving permission blindly, currently all implementation assumes that if the maintainer is not a student, don't give privileges. So returning True in the case not Student, is breaking all those assumptions anyway.

**Comment by GopalGupta8216** [comment]
So what we ultimately want giving privileges to a maintainer who is not a student  i guess, and my code is doing the same and in any case, in order to fullfill our motive those applicatons may needs to change their structure in near future.

**Comment by rish9101** [comment]
See @GopalGupta8216 , passing a named parameter as default permission, in this case, is the goal. Now `get_rights_for_semester` is used to check the rights for many permissions like `helpcentre` , `alohomora` etc. What the actual use-case is that for some of this permission, we might want to give the rights to maintainer(who is not a student) for say `alohomora` but not for `helpcentre`, so in that case,  `alohomora` would just need to pass the extra callback default bool (True) but `helpcentre` need not do it as the default value is False. What your PR does is it simply gives the privilege to a maintainer(who isn't a student) no matter what privilege it's asking for, rather than us deciding if he gets it.

**Comment by GopalGupta8216** [comment]
Okay i got the point that we want to have that power with us that whether we want to give privileges or not according to our needs. Yup that thing was not at all specified clearly from the issue #10 .  I will make another PR by today. @pradumangoyal @rish9101

**Comment by pradumangoyal** [comment]
Don't submit a new PR; push your changes to this branch. The changes will be updated in this PR automatically.

**Comment by GopalGupta8216** [comment]
What about rather than making a new function, we just add another parameter named give_special_privileges default to false and return it in place of true in my code. Is this what you mean by passing callback bool(I am not that good with technical terms ) or is there any other use of having a function rather than a parameter.@pradumangoyal @rish9101

File: utils/rights.py [hunk 1]
```diff
@@ -2,12 +2,13 @@
 from kernel.managers.get_role import get_role
 
 
-def get_rights_for_semester(user, semester_no):
+def get_rights_for_semester(user, semester_no, right_if_not_student=False):
     """
     Check if the given user has enough privileges based on semester number for student maintainers
-    And if the user is a maintainer but not a student than it gives priviles
+    And if the user is a maintainer but not a student than it give privileges according to requirement based on the value of param:right_if_not_student
```
Comment (pradumangoyal) [inline utils/rights.py:8]: Break this line before 80 chars

**Pull request merged**